sectionsUtilityTool/src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionUtilityToolFilter.java

```diff
@@ -0,0 +1,196 @@
+		getExternalAppProperties();
+		
+	}
+
+	@Override
+	public void doFilter(ServletRequest request, ServletResponse response,
+			FilterChain chain) throws IOException, ServletException {
+		M_log.debug("doFilter: Called");
+		HttpServletRequest useRequest = (HttpServletRequest) request;
+		HttpServletResponse useResponse=(HttpServletResponse)response;
+		if(!checkForAuthorization(useRequest)) {
+			useResponse.sendError(403);
+			return;
+		}
+		chain.doFilter(useRequest, response);
+	}
+
+	@Override
+	public void destroy() {
+		M_log.debug("destroy: Called");
+	}
+	
+	protected void getExternalAppProperties() {
+		M_log.debug("getExternalAppProperties(): called");
+		String propertiesFilePathSecure = System.getProperty(SYSTEM_PROPERTY_FILE_PATH_SECURE);
+		String propertiesFilePathLessSecure = System.getProperty(SYSTEM_PROPERTY_FILE_PATH_LESS_SECURE);
+		if (!isEmpty(propertiesFilePathSecure)) {
+			appExtSecurePropertiesFile=PropertiesUtilities.getPropertiesObjectFromURL(propertiesFilePathSecure);
+			if(appExtSecurePropertiesFile!=null) {
+				isTestUrlEnabled = Boolean.parseBoolean(appExtSecurePropertiesFile.getProperty(SectionUtilityToolFilter.PROPERTY_USE_TEST_URL,FALSE));
+			}else {
+				M_log.warn("Failed to load secure application properties from sectionsToolPropsSecure.properties for SectionsTool");
+			}
+			
+		}else {
+			M_log.error("File path for (sectionsToolPropsPathSecure.properties) is not provided");
+		}
+		
+		if (!isEmpty(propertiesFilePathLessSecure)) {
+			appExtLessSecurePropertiesFile=PropertiesUtilities.getPropertiesObjectFromURL(propertiesFilePathLessSecure);
+			if(appExtLessSecurePropertiesFile!=null) {
+				providerURL=appExtLessSecurePropertiesFile.getProperty(PROPERTY_LDAP_SERVER_URL);
+				mcommunityGroup=appExtLessSecurePropertiesFile.getProperty(PROPERTY_AUTH_GROUP);
+			}else {
+				M_log.warn("Failed to load secure application properties from sectionsToolPropsLessSecure.properties for SectionsTool");
+			}
+			
+		}else {
+			M_log.error("File path for (sectionsToolPropsPathLessSecure.properties) is not provided");
+		}
+
+		
+	}
+	 private boolean isEmpty(String value) {
+		return (value == null) || (value.trim().equals(""));
+	}
+	
+	/*
+	 * User is authenticated using cosign and authorized using Ldap. For local development we are enabling
+	 * "testUser" parameter. "testUser" also go through the Ldap authorization process. 
+	 * we have use.test.url configured in the properties file in order to disable usage of "testUser" parameter in PROD.
+	 * 
+	 */
+	private boolean checkForAuthorization(HttpServletRequest request) {
+		M_log.debug("checkLdapForAuthorization(): called");
+     	String remoteUser = request.getRemoteUser();
+     	String testUser = request.getParameter(TEST_USER);
+     	boolean isAuthorized = false;
+		String user=null;
+		
+		String testUserInSession = (String)request.getSession().getAttribute(TEST_USER);
+		
+	    if ( isTestUrlEnabled && testUser != null ) { 
```

Comment:
Minor nit, but please make sure your use of spaces/tabs is consistent so that curly braces line up.